dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -2491,9 +2494,9 @@ public void refreshFeatures(Filter filter) {
         refreshFeatures();
     }
 
-    boolean missingRequiredJava(FeatureResource fr) {
-        Integer requiredJava = fr.getRequireJava();
-        return requiredJava == null ? false : JavaInfo.majorVersion() < requiredJava;
+    public boolean withinJavaRange(FeatureResource fr) {
+        VersionRange range = fr.getJavaRange();
+        return range == null ? true : range.includes(new Version(Integer.toString(JavaInfo.majorVersion())));
```

Comment:
Seems wasteful to put it to a `String` just to parse it again to an `int`.
```suggestion
        return range == null ? true : range.includes(new Version(JavaInfo.majorVersion(), 0, 0));
```
Since it doesn't change, it would be worth caching `new Version(JavaInfo.majorVersion(), 0, 0)` in a private static final variable in this class as well.